presto-docs/pom.xml

```diff
@@ -40,12 +40,12 @@
                 <artifactId>drift-maven-plugin</artifactId>
                 <executions>
                     <execution>
-                        <phase>validate</phase>
+                        <phase>generate-sources</phase>
```

Comment:
can you explain the changes in this file (why did the phases change, and why did "thrift" get added to the directory path?) also, is there any noticeable difference after this change for a regular build?

Reply:
To support out of tree build the `<module>/target` directory has to be linked at the very beginning. The `drift-maven-plugin` is generating `.thrift` file definition into `/target` directory. To make sure the `target` directory is linked before the `drift` plugin invocation It has to be moved to a later build phase. Also `generate-sources` seems to be a more appropriate phase to run a plugin generating code.

Since `PrestoThriftService.thrift` been moved to a later phase I had to move the processing as well (to `process-sources` phase).

I had to move `run-sphinx` plugging to compile phase as keeping it in `package` phase introduces non determinism. Plugins in the same phase are invoked in order. But order of plugins in Maven is somehow obscure (for example what if a child module has plugins in different order than the parent module).

The reason why `/thrift` is needed is funny. Basically the `thrift` generator plugin tries to create parent directories automatically. And if a parent director is a symbolic link - it fails.